src/instructlab/utils.py

```diff
@@ -44,6 +44,38 @@ class TaxonomyReadingException(Exception):
     """An exception raised during reading of the taxonomy."""
 
 
+class Message(TypedDict):
+    """
+    Represents a message within an AI conversation.
+    """
+
+    content: str
+    # one of: "user", "assistant", or "system"
+    role: str
```

Comment:
Can you do an enum here then?

```
from enum import Enum

class Role(Enum):
    USER = "user"
    ASSISTANT = "assistant"
    SYSTEM = "system"

class Message(TypedDict):
    ...
    role: Role
```

Reply:
We've discussed this in a previous conversation which you can read here: https://github.com/instructlab/instructlab/pull/1644#discussion_r1671962251

The TLDR is that while an Enum helps us narrow down the values, it adds too much additional complexity and headache when a simple `str` would serve the job well.